### PR TITLE
add vscode/docker interactive tutorials

### PR DESCRIPTION
**Pull request by craigshoemaker**
Add interactive tutorials for Node.js apps using Visual Studio to deploy to Azure App Service. This PR includes tutorials for both zip deploy and custom containers.

Requesting review from @mmacy and @fiveisprime

### CODE REVIEW

**Comment by mikekinsman** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/node-vscode-deployment-zip.yml](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/node-vscode-deployment-zip.yml) | :x:Error | | [Details](#user-content-docs-ref-conceptual%2Fnode-vscode-deployment-zip.yml)
[]() | :x:Error | | [Details](#user-content-)

<a name="docs-ref-conceptual%2Fnode-vscode-deployment-zip.yml"></a>
### [docs-ref-conceptual/node-vscode-deployment-zip.yml](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/node-vscode-deployment-zip.yml)
 - **[Error]** Unable to load file: docs-ref-conceptual/node-vscode-deployment-zip.yml via processor: YamlDocumentProcessor.

<a name=""></a>
### []()
 - **[Error]** YamlDotNet.Core.SyntaxErrorException: (Line: 9, Col: 21, Idx: 404) - (Line: 9, Col: 22, Idx: 405): While scanning an anchor or alias, did not find expected alphabetic or numeric character.
   at Microsoft.DocAsCode.Build.Engine.SingleDocumentBuilder.GetInnerContexts(DocumentBuildParameters parameters, IEnumerable`1 processors, TemplateProcessor templateProcessor, IHostServiceCreator creator)
   at Microsoft.DocAsCode.Build.Engine.SingleDocumentBuilder.BuildCore(DocumentBuildParameters parameters)
   at Microsoft.DocAsCode.Build.Engine.DocumentBuilder.BuildCore(DocumentBuildParameters parameter, IMarkdownServiceProvider markdownServiceProvider, BuildInfo currentBuildInfo, BuildInfo lastBuildInfo)
   at Microsoft.DocAsCode.Build.Engine.DocumentBuilder.Build(IList`1 parameters, String outputDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument(BuildJsonConfig config, TemplateManager templateManager, String baseDirectory, String outputDirectory, String pluginDirectory, String templateDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument()

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C12%5C5%5Cb6261451-6613-d51e-95b5-8b5dc96f3a2c%5CPullRequest%5C201712052157299876-236%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=JUUrjqbSMNIcOsPBBJSbSzPR8lX3TC%2FUMU11i8uDaVc%3D&st=2017-12-05T21%3A57%3A02Z&se=2018-01-05T22%3A02%3A02Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by mikekinsman** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/node-vscode-deployment-zip.yml](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/node-vscode-deployment-zip.yml) | :x:Error | | [Details](#user-content-docs-ref-conceptual%2Fnode-vscode-deployment-zip.yml)
[]() | :x:Error | | [Details](#user-content-)

<a name="docs-ref-conceptual%2Fnode-vscode-deployment-zip.yml"></a>
### [docs-ref-conceptual/node-vscode-deployment-zip.yml](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/node-vscode-deployment-zip.yml)
 - **[Error]** Unable to load file: docs-ref-conceptual/node-vscode-deployment-zip.yml via processor: YamlDocumentProcessor.

<a name=""></a>
### []()
 - **[Error]** YamlDotNet.Core.SyntaxErrorException: (Line: 10, Col: 22, Idx: 445) - (Line: 10, Col: 23, Idx: 446): While scanning an anchor or alias, did not find expected alphabetic or numeric character.
   at Microsoft.DocAsCode.Build.Engine.SingleDocumentBuilder.GetInnerContexts(DocumentBuildParameters parameters, IEnumerable`1 processors, TemplateProcessor templateProcessor, IHostServiceCreator creator)
   at Microsoft.DocAsCode.Build.Engine.SingleDocumentBuilder.BuildCore(DocumentBuildParameters parameters)
   at Microsoft.DocAsCode.Build.Engine.DocumentBuilder.BuildCore(DocumentBuildParameters parameter, IMarkdownServiceProvider markdownServiceProvider, BuildInfo currentBuildInfo, BuildInfo lastBuildInfo)
   at Microsoft.DocAsCode.Build.Engine.DocumentBuilder.Build(IList`1 parameters, String outputDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument(BuildJsonConfig config, TemplateManager templateManager, String baseDirectory, String outputDirectory, String pluginDirectory, String templateDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument()

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C12%5C6%5Cb6261451-6613-d51e-95b5-8b5dc96f3a2c%5CPullRequest%5C201712061100357659-236%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=OnF3S0fFgyPENPJZ%2FNPlriI1E3wZVeBGRafY%2BdIWSKM%3D&st=2017-12-06T10%3A59%3A47Z&se=2018-01-06T11%3A04%3A47Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by mikekinsman** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/node-vscode-deployment-container.yml](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/node-vscode-deployment-container.yml) | :x:Error | | [Details](#user-content-docs-ref-conceptual%2Fnode-vscode-deployment-container.yml)
[]() | :x:Error | | [Details](#user-content-)

<a name="docs-ref-conceptual%2Fnode-vscode-deployment-container.yml"></a>
### [docs-ref-conceptual/node-vscode-deployment-container.yml](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/node-vscode-deployment-container.yml)
 - **[Error]** Unable to load file: docs-ref-conceptual/node-vscode-deployment-container.yml via processor: YamlDocumentProcessor.

<a name=""></a>
### []()
 - **[Error]** YamlDotNet.Core.SyntaxErrorException: (Line: 10, Col: 22, Idx: 471) - (Line: 10, Col: 23, Idx: 472): While scanning an anchor or alias, did not find expected alphabetic or numeric character.
   at Microsoft.DocAsCode.Build.Engine.SingleDocumentBuilder.GetInnerContexts(DocumentBuildParameters parameters, IEnumerable`1 processors, TemplateProcessor templateProcessor, IHostServiceCreator creator)
   at Microsoft.DocAsCode.Build.Engine.SingleDocumentBuilder.BuildCore(DocumentBuildParameters parameters)
   at Microsoft.DocAsCode.Build.Engine.DocumentBuilder.BuildCore(DocumentBuildParameters parameter, IMarkdownServiceProvider markdownServiceProvider, BuildInfo currentBuildInfo, BuildInfo lastBuildInfo)
   at Microsoft.DocAsCode.Build.Engine.DocumentBuilder.Build(IList`1 parameters, String outputDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument(BuildJsonConfig config, TemplateManager templateManager, String baseDirectory, String outputDirectory, String pluginDirectory, String templateDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument()

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C12%5C6%5Cb6261451-6613-d51e-95b5-8b5dc96f3a2c%5CPullRequest%5C201712061150091520-236%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=5NvNvWOCRovdjwBHLKdxbJ8%2Fr8pvqZvcLF9IV4oHeX0%3D&st=2017-12-06T11%3A46%3A19Z&se=2018-01-06T11%3A51%3A19Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs-ref-conceptual/node-vscode-deployment-zip.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    - [Visual Studio Code](https://code.visualstudio.com/download)
+    - [Docker](https://www.docker.com/)
+    - [Node Pack for Azure](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-node-azure-pack) for Visual Studio Code
+- title: Create a new application
+  durationInMinutes: 2
+  content: |
+    From the Visual Studio Code **App Service** extension, click the `+` button to create a new application. Once clicked, the editor displays a series of prompts at the top of the screen. 
+
+    > [!NOTE]
+    > Note: To observe the status as Azure creates your application, you can open Visual Studio Code's Integrated Terminal through the `View > Integrated Terminal` menu option.
```
Comment (mmacy) [inline docs-ref-conceptual/node-vscode-deployment-zip.yml]: I haven't seen one of these interactive tutorials rendered, but "Note" is added the note box header automatically on docs.ms.com. As such, recommend striking "Note:" from this note.

**Comment by mikekinsman** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[.openpublishing.publish.config.json](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/.openpublishing.publish.config.json) | :white_check_mark:Succeeded | |
[docs-ref-conceptual/node-vscode-deployment-container.yml](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/node-vscode-deployment-container.yml) | :white_check_mark:Succeeded | |
[docs-ref-conceptual/node-vscode-deployment-zip.yml](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/node-vscode-deployment-zip.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C12%5C7%5Cb6261451-6613-d51e-95b5-8b5dc96f3a2c%5CPullRequest%5C201712071530165950-236%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=gkvyc%2BA1vIcl71m9W4XQaTcZcgpxpXFpVYvecxYq%2BZg%3D&st=2017-12-07T15%3A43%3A23Z&se=2018-01-07T15%3A48%3A23Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by mikekinsman** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[.openpublishing.publish.config.json](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/.openpublishing.publish.config.json) | :white_check_mark:Succeeded | |
[docs-ref-conceptual/media/vscode-deployment/vscode-deployment-add-docker-files.png](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/media/vscode-deployment/vscode-deployment-add-docker-files.png) | :white_check_mark:Succeeded | |
[docs-ref-conceptual/media/vscode-deployment/vscode-deployment-add-platform.png](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/media/vscode-deployment/vscode-deployment-add-platform.png) | :white_check_mark:Succeeded | |
[docs-ref-conceptual/media/vscode-deployment/vscode-deployment-docker-build.png](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/media/vscode-deployment/vscode-deployment-docker-build.png) | :white_check_mark:Succeeded | |
[docs-ref-conceptual/media/vscode-deployment/vscode-deployment-port.png](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/media/vscode-deployment/vscode-deployment-port.png) | :white_check_mark:Succeeded | |
[docs-ref-conceptual/node-vscode-deployment-container.yml](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/node-vscode-deployment-container.yml) | :white_check_mark:Succeeded | |
[docs-ref-conceptual/node-vscode-deployment-zip.yml](https://github.com/craigshoemaker/azure-docs-sdk-node/blob/vscode-containers-interactive/docs-ref-conceptual/node-vscode-deployment-zip.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C12%5C13%5Cb6261451-6613-d51e-95b5-8b5dc96f3a2c%5CPullRequest%5C201712131454544403-236%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=a59QWoFPYgu03ziCeBU8F5%2Fr%2FNE5JGtnJNKNqQx6C7g%3D&st=2017-12-13T15%3A00%3A30Z&se=2018-01-13T15%3A05%3A30Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by craigshoemaker** [comment]
Closing this PR as @fiveisprime is taking on this content.

**Pull request closed without merging**